service/history/service.go

```diff
@@ -107,30 +110,31 @@ func NewConfig(dc *dynamicconfig.Collection, numberOfShards int) *Config {
 		TimerTaskMaxRetryCount:                             5,
 		TimerProcessorGetFailureRetryCount:                 5,
 		TimerProcessorCompleteTimerFailureRetryCount:       10,
-		TimerProcessorUpdateAckInterval:                    10 * time.Second,
-		TimerProcessorForceUpdateInterval:                  10 * time.Minute,
+		TimerProcessorUpdateShardTaskCount:                 100,
+		TimerProcessorUpdateAckInterval:                    1 * time.Minute,
 		TimerProcessorCompleteTimerInterval:                1 * time.Second,
 		TimerProcessorMaxPollInterval:                      60 * time.Second,
 		TimerProcessorStandbyTaskDelay:                     0 * time.Minute,
 		TransferTaskBatchSize:                              10,
 		TransferProcessorMaxPollRPS:                        100,
-		TransferProcessorMaxPollInterval:                   60 * time.Second,
-		TransferProcessorUpdateAckInterval:                 10 * time.Second,
-		TransferProcessorForceUpdateInterval:               10 * time.Minute,
-		TransferProcessorCompleteTransferInterval:          1 * time.Second,
-		TransferProcessorStandbyTaskDelay:                  0 * time.Minute,
 		TransferTaskWorkerCount:                            10,
 		TransferTaskMaxRetryCount:                          100,
 		TransferProcessorCompleteTransferFailureRetryCount: 10,
+		TransferProcessorUpdateShardTaskCount:              100,
+		TransferProcessorMaxPollInterval:                   60 * time.Second,
+		TransferProcessorUpdateAckInterval:                 1 * time.Minute,
+		TransferProcessorCompleteTransferInterval:          1 * time.Second,
```

Comment:
1 second for complete sounds pretty low, considering we have 16K shards.  I think it should also be notification based when there is something to cleanup.  
Let's file a task to review all service configs.

Reply:
#741